overlord/snapstate/snapstate.go

```diff
@@ -1607,6 +1616,14 @@ func Remove(st *state.State, name string, revision snap.Revision) (*state.TaskSe
 			si := seq[i]
 			addNext(removeInactiveRevision(st, name, si.Revision))
 		}
+
+		if info.SnapID != "" {
```

Comment:
If you follow my advice elsewhere this could perhaps be triggered not by the id being present but rather by the flag being set in the state.